geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/AttributeIdxStrategy.scala

```diff
@@ -161,13 +165,21 @@ trait AttributeIdxStrategy extends Strategy with Logging {
       classOf[AttributeIndexIterator]
     )
 
+    val coverage = featureType.getDescriptor(attributeName).getIndexCoverage()
+
     configureFeatureTypeName(cfg, featureType.getTypeName)
     configureFeatureEncoding(cfg, encoding)
-    configureStFilter(cfg, stFilter)
-    configureEcqlFilter(cfg, ecqlFilter.map(ECQL.toCQL))
-    configureAttributeName(cfg, attributeName)
-    configureIndexCoverage(cfg, featureType.getDescriptor(attributeName).getIndexCoverage())
     configureIndexValues(cfg, featureType)
+    configureAttributeName(cfg, attributeName)
+    configureIndexCoverage(cfg, coverage)
+    configureVersion(cfg, version)
+    if (coverage == IndexCoverage.FULL) {
+      // combine filters into one check
+      configureEcqlFilter(cfg, filterListAsAnd(Seq(stFilter ++ ecqlFilter).flatten).map(ECQL.toCQL))
```

Comment:
For capture, I'd suggest cut and pasting some of this conversation into a JIRA ticket.  I have been suggesting adding comments in the code which link to the ticket.

If you meant capturing info about the query planner, I suggest creating versioned documentation.  The query explainer should spit out the metadata which could be used, and then move on to explaining the plan.  

In general, I imagine there will be a 'covering index' trait for the ST and ATTR tables which will help organize this bit of the query planning.